src/test/java/dog/pawbook/logic/commands/EnrolCommandTest.java

```diff
@@ -0,0 +1,152 @@
+package dog.pawbook.logic.commands;
+
+import static dog.pawbook.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static dog.pawbook.testutil.Assert.assertThrows;
+import static dog.pawbook.testutil.TypicalEntities.getDatabaseWithPrograms;
+import static dog.pawbook.testutil.TypicalId.ID_FIFTEEN;
+import static dog.pawbook.testutil.TypicalId.ID_FOUR;
+import static dog.pawbook.testutil.TypicalId.ID_SIX;
+import static dog.pawbook.testutil.TypicalId.ID_SIXTEEN;
+import static dog.pawbook.testutil.TypicalId.ID_TWO;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import dog.pawbook.model.Model;
+import dog.pawbook.model.ModelManager;
+import dog.pawbook.model.UserPrefs;
+import dog.pawbook.model.managedentity.IdMatchPredicate;
+import dog.pawbook.model.managedentity.program.Program;
+
+public class EnrolCommandTest {
+    private Model model;
+    private Model expectedModel;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getDatabaseWithPrograms(), new UserPrefs());
+        expectedModel = new ModelManager(model.getDatabase(), new UserPrefs());
+    }
+
+    @Test
+    public void constructor_nullEnrol_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new EnrolCommand(null, null));
+    }
+
+    @Test
+    public void execute_enrolOneDogOneProgram_successful() {
+        Set<Integer> dogIdSet = new HashSet<>();
+        Set<Integer> programIdSet = new HashSet<>();
+        dogIdSet.add(ID_TWO);
+        programIdSet.add(ID_FIFTEEN);
+
+        Program editedProgram = (Program) expectedModel.getEntity(15);
```

Comment:
hmm maybe consistent use of the constants? avoid the literals since you have defined the literals

Reply:
omg my bad LOL i think i was doing some debugging and changing back and forth